### PR TITLE
[MENFORCER-313] - Added [<scope>] to the conflicting artifacts in the output. Highlight those with compile scope.

### PR DESCRIPTION
**Pull request by belingueres**
[MENFORCER-313] - Added [<scope>] to the conflicting artifacts in the output.
Highlight those with compile scope for easy reading.

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MENFORCER) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MENFORCER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MENFORCER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: maven-enforcer-plugin/src/it/projects/require-upper-bound-dependencies-managed_failure/verify.groovy [hunk 1]
```diff
@@ -21,9 +21,9 @@ File buildLog = new File( basedir, 'build.log' )
 assert buildLog.text.contains( 'Rule 0: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:' )
 
 def message = 'Failed while enforcing RequireUpperBoundDeps. The error(s) are [' + LS+
-'Require upper bound dependencies error for org.apache.maven.plugins.enforcer.its:menforcer146-x:1.1 paths to dependency are:' + LS+
+'Require upper bound dependencies error for org.apache.maven.plugins.enforcer.its:menforcer146-x:1.1 [compile] paths to dependency are:' + LS+
```
Comment (rfscholte) [inline maven-enforcer-plugin/src/it/projects/require-upper-bound-dependencies-managed_failure/verify.groovy]: The word compile looks weird here.

File: enforcer-rules/src/main/java/org/apache/maven/plugins/enforcer/RequireUpperBoundDeps.java [hunk 2]
```diff
@@ -236,7 +237,17 @@ private String getFullArtifactName( DependencyNode node, boolean usePremanaged )
         {
             version = uniqueVersions ? artifact.getVersion() : artifact.getBaseVersion();
         }
-        return artifact.getGroupId() + ":" + artifact.getArtifactId() + ":" + version;
+        String result = artifact.getGroupId() + ":" + artifact.getArtifactId() + ":" + version;
+        if ( artifact.getScope() != null )
```
Comment (rfscholte) [inline enforcer-rules/src/main/java/org/apache/maven/plugins/enforcer/RequireUpperBoundDeps.java]: Based on the eclipse output you only want the conflicts to be highlighted, not the entire tree.
I'd advice to add a screenshot of the result to ensure we're working towards the same result.

Comment (belingueres) [reply enforcer-rules/src/main/java/org/apache/maven/plugins/enforcer/RequireUpperBoundDeps.java]: requireUpper

Eclipse show in black font the compile dependencies and in grey color other scopes (test, runtime). I added the font highlight to the compile dependencies to resemble this, but I'm open to suggestions.

File: maven-enforcer-plugin/src/it/projects/require-upper-bound-dependencies-managed_failure/verify.groovy (hunk 1 above)
Comment (belingueres) [reply maven-enforcer-plugin/src/it/projects/require-upper-bound-dependencies-managed_failure/verify.groovy]: Yes, adding the scope there it does look out of place.

Comment (belingueres) [reply maven-enforcer-plugin/src/it/projects/require-upper-bound-dependencies-managed_failure/verify.groovy]: modified, rebased and squashed.

**Comment by jira-importer** [comment]
Resolve #795

**Pull request closed without merging**